service/src/main/kotlin/fi/espoo/evaka/placement/PlacementPlanService.kt

```diff
@@ -47,10 +47,12 @@ class PlacementPlanService(
 
         when (application.type) {
             ApplicationType.PRESCHOOL -> {
-                val term = derivePreschoolTerm(startDate)
-                val period = FiniteDateRange(startDate, term.end)
+                val preschoolTerms = tx.getActivePreschoolTermAt(startDate)
+                    ?: throw Exception("No suitable preschool term found for start date $startDate")
+                val exactTerm = if (isSvebiUnit(tx, preferredUnits[0].id)) preschoolTerms.swedishPreschool else preschoolTerms.finnishPreschool
+                val period = FiniteDateRange(startDate, exactTerm.end)
                 val preschoolDaycarePeriod = if (type == PlacementType.PRESCHOOL_DAYCARE || type == PlacementType.PREPARATORY_DAYCARE) {
-                    FiniteDateRange(startDate, LocalDate.of(term.end.year, 7, 31))
+                    FiniteDateRange(startDate, LocalDate.of(preschoolTerms.extendedTerm.end.year, 7, 31))
```

Comment:
hmm why do we hard code 31.7.

Reply:
ah nevermind, I guess that is not available in the data

Reply:
I'll ask around to figure how that end date should be defined but will merge this now anyway